src/loop.c

```diff
@@ -125,6 +151,7 @@ long long us_loop_iteration_number(struct us_loop_t *loop) {
 /* These may have somewhat different meaning depending on the underlying event library */
 void us_internal_loop_pre(struct us_loop_t *loop) {
     loop->data.iteration_nr++;
+    us_internal_handle_low_priority_sockets(loop);
```

Comment:
What if we have sockets in low prio list, yet we have no more events - then next iteration will never happen and we have some kind of locked situation?

Reply:
I don't think so ... as long as `us_internal_handle_low_priority_sockets()` guarantees, that it always puts some sockets out of the low-priority-queue on every run, as long it is not empty (which should be the case currently).

Okay, while thinking about your comment, there is indeed one precondition: As we only put sockets into the low-priority-queue, which are definitely readable, it should not be a problem ... but we assume, that a socket that was readable some iterations before, is still readable when we tell epoll again, that we are interested. But I don't know of any situation where this can be a problem, as we do not do anything with this socket in the meantime.

The only thing that can happen in the meantime, is that the socket is closed. And this case is handled here:

https://github.com/uNetworking/uSockets/pull/97/files#diff-ab341bde26e91c15deea4b1904d90637R97

Reply:
No I mean what if you get 6 incoming SSL handshakes - 5 are handled this iteration immediately but the last 6th one is put into low priority list, where it is fetched from _next iteration_. But there is nothing guaranteeing that there will ever be another iteration (well we have a timer that ticks every 4 seconds but if we remove that timer you could end up with sockets stuck in the low prio list).

Reply:
What I mean is, we probably need to call us_wakeup_loop to force at least one event to drive a new iteration?

Reply:
That should not be a problem I think ... the loop iteration will look like this:

- epoll() says "6 ready connections"
- we handle 5 immediately
- we remove READABLE poll from socket No. 6
- the global while-loop continues from top
- us_internal_loop_pre() will be called
- we add READABLE poll to socket No. 6 there
- epoll() will be called
- we handle socket No. 6

So I do not see any locked situation. Or do I miss something important?